zkevm-circuits/src/witness/rw.rs

```diff
@@ -561,69 +574,43 @@ impl Rw {
-                    rlc::value(&value.to_le_bytes(), randomness)
-                }
-                AccountFieldTag::Nonce | AccountFieldTag::NonExisting => value.to_scalar().unwrap(),
-            },
+            Self::Start { .. } => WordNew::default(),
+            Self::CallContext { value, .. } => WordNew::from_u256(*value),
+            Self::Account { value, .. } => WordNew::from_u256(*value),
             Self::AccountStorage { value, .. } | Self::Stack { value, .. } => {
-                rlc::value(&value.to_le_bytes(), randomness)
+                WordNew::from_u256(*value)
             }
-
-            Self::TxLog {
-                field_tag, value, ..
-            } => match field_tag {
-                TxLogFieldTag::Topic => rlc::value(&value.to_le_bytes(), randomness),
-                _ => value.to_scalar().unwrap(),
-            },
+            Self::TxLog { value, .. } => WordNew::from_u256(*value),
 
             Self::TxAccessListAccount { is_warm, .. }
-            | Self::TxAccessListAccountStorage { is_warm, .. } => F::from(*is_warm as u64),
-            Self::Memory { byte, .. } => F::from(u64::from(*byte)),
-            Self::TxRefund { value, .. } | Self::TxReceipt { value, .. } => F::from(*value),
+            | Self::TxAccessListAccountStorage { is_warm, .. } => {
+                WordNew::new([F::from(*is_warm as u64), F::ZERO])
+            }
+            Self::Memory { byte, .. } => WordNew::new([F::from(u64::from(*byte)), F::ZERO]),
+            Self::TxRefund { value, .. } | Self::TxReceipt { value, .. } => {
+                WordNew::new([F::from(*value), F::ZERO])
+            }
         }
     }
 
-    pub(crate) fn value_prev_assignment<F: Field>(&self, randomness: F) -> Option<F> {
+    pub(crate) fn value_prev_assignment<F: Field>(&self) -> Option<WordNew<F>> {
         match self {
-            Self::Account {
-                value_prev,
-                field_tag,
-                ..
-            } => Some(match field_tag {
-                AccountFieldTag::CodeHash | AccountFieldTag::Balance => {
-                    rlc::value(&value_prev.to_le_bytes(), randomness)
-                }
-                AccountFieldTag::Nonce | AccountFieldTag::NonExisting => {
-                    value_prev.to_scalar().unwrap()
-                }
-            }),
-            Self::AccountStorage { value_prev, .. } => {
-                Some(rlc::value(&value_prev.to_le_bytes(), randomness))
-            }
+            Self::Account { value_prev, .. } => Some(WordNew::from_u256(*value_prev)),
+            Self::AccountStorage { value_prev, .. } => Some(WordNew::from_u256(*value_prev)),
             Self::TxAccessListAccount { is_warm_prev, .. }
             | Self::TxAccessListAccountStorage { is_warm_prev, .. } => {
-                Some(F::from(*is_warm_prev as u64))
+                Some(WordNew::new([F::from(*is_warm_prev as u64), F::ZERO]))
```

Comment:
Can the from_u64 constructor work here?
```suggestion
                Some(WordNew::from_u64(*is_warm_prev as u64))
```

Reply:
fixed in latest commit https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1441/commits/f4cc21fb96ebece18e5b9cbf656ee63b94a3c609